yourbench/pipeline/single_shot_question_generation.py

```diff
@@ -308,35 +308,39 @@ def _process_responses_and_build_dataset(
 
             # Otherwise, process each QA pair
             for pair in qa_pairs:
-                # Safely extract data from pair
-                question_text = str(pair.get("question", "")).strip()
-                answer_text = str(pair.get("answer", "")).strip()
-                difficulty_val = _force_int_in_range(pair.get("estimated_difficulty", 5), 1, 10)
-                question_type = str(pair.get("question_type", "unknown"))
-                thought_process = str(pair.get("thought_process", ""))
-                citations = pair.get("citations", [])
-                if not isinstance(citations, list):
-                    citations = []
-
-                if not question_text:
-                    logger.debug(f"Empty question found; skipping this QA pair (row_index={row_index}).")
+                try:
+                    # Safely extract data from pair
+                    question_text = str(pair.get("question", "")).strip()
+                    answer_text = str(pair.get("answer", "")).strip()
+                    difficulty_val = _force_int_in_range(pair.get("estimated_difficulty", 5), 1, 10)
+                    question_type = str(pair.get("question_type", "unknown"))
+                    thought_process = str(pair.get("thought_process", ""))
+                    citations = pair.get("citations", [])
+                    if not isinstance(citations, list):
+                        citations = []
+
+                    if not question_text:
+                        logger.debug(f"Empty question found; skipping this QA pair (row_index={row_index}).")
```

Comment:
We would want to log also the associated model + doc + chunk if we have them (to help people debug more easily)